django/test/signals.py

```diff
@@ -178,3 +178,10 @@ def static_finders_changed(**kwargs):
     }:
         from django.contrib.staticfiles.finders import get_finder
         get_finder.cache_clear()
+
+
+@receiver(setting_changed)
+def auth_user_model_changed(**kwargs):
+    if kwargs['setting'] == 'AUTH_USER_MODEL':
+        from django.apps import apps
+        apps.get_swappable_setting.cache_clear()
```

Comment:
This bit of code implies that you can change `AUTH_USER_MODEL` and Django will deal with it. I don't think that's true.

Changing `AUTH_USER_MODEL` at runtime would require rebuilding the entier app registry. Think about the common pattern `ForeignKey(settings.AUTH_USER_MODEL, ...)` for example.